src/modules/0.0.22/aws_rds/index.ts

```diff
@@ -443,33 +457,47 @@ class ParameterGroupMapper extends MapperBase<ParameterGroup> {
           result?.DBParameterGroupName ?? '',
         );
         // We map this into the same kind of entity as `obj`
-        const newEntity = this.parameterGroupMapper(newObject);
+        const newEntity = this.parameterGroupMapper(newObject, e.region);
+        newEntity.id = e.id;
         // Save the record back into the database to get the new fields updated
         await this.module.parameterGroup.db.update(newEntity, ctx);
         out.push(newEntity);
       }
       return out;
     },
     read: async (ctx: Context, id?: string) => {
-      const client = (await ctx.getAwsClient()) as AWS;
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
       if (id) {
-        const parameterGroup = await this.getDBParameterGroup(client.rdsClient, id);
+        const { name, region } = this.idFields(id);
+        const client = (await ctx.getAwsClient(region)) as AWS;
+        const parameterGroup = await this.getDBParameterGroup(client.rdsClient, name);
         if (!parameterGroup) return;
-        return this.parameterGroupMapper(parameterGroup);
+        return this.parameterGroupMapper(parameterGroup, region);
       } else {
-        const parameterGroups = await this.getDBParameterGroups(client.rdsClient);
         const out = [];
-        for (const pg of parameterGroups) {
-          out.push(this.parameterGroupMapper(pg));
+        for (const region of enabledRegions) {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          const parameterGroups = await this.getDBParameterGroups(client.rdsClient);
+          for (const pg of parameterGroups) {
+            out.push(this.parameterGroupMapper(pg, region));
+          }
```

Comment:
let's do this in parallel with `await.Promise.all()` like here https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_api_gateway/index.ts#L79L91